docs/source/370_jupyter_lmod.rst

```diff
@@ -0,0 +1,130 @@
+Jupyter and Lmod
+================
+
+It is possible to use Lmod inside of Jupyter with the terminal or jupyter-lmod.
+
+jupyter-lmod is a frontend and server extension for Jupyter (notebook and jupyterlab).
+The extension graphically provides most of Lmod's functions:
+    - Browser and filter available modules
+    - Load and unload modules
+    - Display a module content
+    - Create and load collections
+
+The extension is located in the 'Softwares' tab of the main interface in Jupyter Notebook,
+while in JupyterLab, it can be accessed as a sidepanel by clicking on the Lmod logo icon
+located in the sidebar.
+
+Installation
+~~~~~~~~~~~~
+
+To install jupyter-lmod extension: ::
+
+    pip install jupyterlmod
+
+This enables the extension for both Jupyter Notebook and JupyterLab.
+
+How does it work?
+~~~~~~~~~~~~~~~~~
+
+jupyter-lmod uses lmod python interface to generate Python code
+that modifies the environment variables of the Python process running Jupyter. All child
+processes of Jupyter (including kernels) created after Lmod calls inherit
+the environment variables as defined by the set of loaded modules.
+
+In concrete terms, kernels in use by active notebooks will not be aware of changes
+made to the environment (e.g. **$PATH**, **$PYTHONPATH**, **$LD_LIBRARY_PATH**) by modules
+loaded after they were spawned. Depending on the Jupyter version, a manual kernel restart
+or a complete kernel shutdown/start cycle is needed to make an active notebook aware of changes
+to the environment variables.
+
+Lmod environment variable configuration for jupyter-lmod
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The plugin requires the environment variable **$LMOD_CMD** to be defined when Jupyter starts,
+plus all environment variables required by Lmod to work properly (i.e.: **$MODULEPATH**).
+
+Some JupyterHub spawners do not setup the complete user's environment, leaving **$LMOD_CMD**
+and **$MODULEPATH** undefined. To address this issue, the environment variables have to be
+defined in **jupyterhub_config.py** using
+`c.Spawner.environment <https://jupyterhub.readthedocs.io/en/stable/api/spawner.html#jupyterhub.spawner.Spawner.environment>`_.
+
+If the environment variables are defined before JupyterHub launch, you can add their
+names only to `c.Spawner.env_keep <https://jupyterhub.readthedocs.io/en/stable/api/spawner.html#jupyterhub.spawner.Spawner.env_keep>`_
+instead.
```

Comment:
Technically these variable are only needed in the single-user server. JupyterHub itself does not need them.
```suggestion
If the environment variables are already defined before JupyterHub is launched
or before the single-user server starts (Jupyter Notebook or JupyterLab),
you can propagate these variable by adding their names to
`c.Spawner.env_keep <https://jupyterhub.readthedocs.io/en/stable/api/spawner.html#jupyterhub.spawner.Spawner.env_keep>`_ instead.
```